napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,521 @@
+from copy import deepcopy
+from typing import Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+from pydantic import root_validator, validator
+
+from ...utils.colormaps import Colormap
+from ...utils.colormaps.categorical_colormap import CategoricalColormap
+from ...utils.colormaps.colormap_utils import ensure_colormap
+from ...utils.events import EventedModel
+from ...utils.events.custom_types import Array
+from ._color_manager_constants import ColorMode
+from .color_manager_utils import (
+    _validate_colormap_mode,
+    _validate_cycle_mode,
+    guess_continuous,
+    is_color_mapped,
+)
+from .color_transformations import (
+    ColorType,
+    normalize_and_broadcast_colors,
+    transform_color,
+    transform_color_with_defaults,
+)
+
+
+class ColorProperties(EventedModel):
+    name: str
+    values: np.ndarray
+    current_value: Optional[Any] = None
+
+
+class ColorManager(EventedModel):
```

Comment:
General comment, I wonder if we can make some more of the `add`, `remove`, `update_current_properties `, etc methods private? Which of these would a notebook/ scripting user actually want to use vs are convenience methods for us? I flagged the ones I thought could most easily be private.

Reply:
I think this is a good point. I've made the methods the user should directly interact with private.